ls1intum/Artemis
# `Assessment`: Improve user interface for bonus and grading keys

### Checklist
#### General


#### Client

### Motivation and Context


The Grading Key and Bonus configuration pages have some usability issues such as non-uniform vertical alignments and unintuitive placement of buttons. These can be confusing to users and lead to a suboptimal user experience.

### Description

Improvements made in this PR are listed below:
- Grading Key
  - Moved common buttons (generate default, import-export) above the Interval/Detailed tabs.
  - Display the title of the exam or course instead of the id.
  - Aligned import-export buttons to the right to make the layout similar to other course/exam management pages.
  - Fixed vertical alignments of elements on the leftmost and rightmost sides.
  - Used fewer font sizes and boldness (Made Interval/Detailed tabs not bold).
  - Set table vertical alignment to middle.

- Bonus
  - Add numbering before each action item.
  - Tried to emphasize help text for inputs are extra information (and user can skip reading if not needed) by adjusting font sizes and moving the help icon to the beginning of the help message on create mode.
  - Convert upper and title cases to lowercase.
  - Fixed vertical alignments of elements on the leftmost side.
  - Improved font sizes and boldness.
  - Replaced some divs and spans with more semantic elements.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exam
- 1 Course (optional)

1. Log in to Artemis
2. Navigate to Course Administration
3. Go to the `Grading Key` page of a course or exam.
4. Observe the UI changes written in the Description section above and make sure functionality is working as before.
5. Go to the `Bonus` page of an exam.
6. Observe the UI changes written in the Description section above and make sure functionality is working as before.

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage




Not changed.

### Screenshots



#### Bonus Create - Old
![bonus_create_old](https://user-images.githubusercontent.com/18427209/210798826-583b0506-a47c-42c4-a933-10fb63559e12.png) 

#### Bonus Create - New
![bonus_create_new_20220105](https://user-images.githubusercontent.com/18427209/210800743-60b619da-c5b4-481f-9fa4-a28273c3fd8c.png)

#### Bonus Edit - Old
![bonus_edit_old](https://user-images.githubusercontent.com/18427209/210798866-02cf8951-615d-40cb-995c-1cde8e020059.png)

#### Bonus Edit - New
![bonus_edit_new_20220105](https://user-images.githubusercontent.com/18427209/210800559-d3f42225-0e0c-4c8a-be86-59007c8225da.png)

#### Grading Key - Old
![gradingKey_interval_old](https://user-images.githubusercontent.com/18427209/210799128-7b799fec-2b85-49fd-9c3d-1f3945be17b5.png)

#### Grading Key - New
![gradingKey_interval_new_20220105](https://user-images.githubusercontent.com/18427209/210799979-1d71cc5d-34be-4895-9edc-45ff002f5b4a.png)
